components/renku_data_services/users/db.py

```diff
@@ -176,36 +177,45 @@ async def _get_user(self, id: str) -> UserInfo | None:
     @dispatch_message(events.UpdateOrInsertUser)
     async def update_or_insert_user(
         self, user_id: str, payload: dict[str, Any], *, session: AsyncSession | None = None
-    ) -> UserWithNamespaceUpdate:
+    ) -> UserInfoUpdate:
         """Update a user or insert it if it does not exist."""
         if not session:
             raise errors.ProgrammingError(message="A database session is required")
-        res = await session.execute(select(UserORM).where(UserORM.keycloak_id == user_id))
+        res = await session.execute(
+            select(UserORM).where(UserORM.keycloak_id == user_id).options(selectinload(UserORM.namespace))
+        )
```

Comment:
suggestion (non-blocking): This is really nit-picky and really not a problem. But it may be nicer to add this on the schema so that you dont have to repeat `options(selectinload(UserORM.namespace))`. But it is not that big of a deal. So ignore this if you wish.